### PR TITLE
Add unit test to cover hash code operator in REAL type

### PR DESCRIPTION
**Pull request by jessesleeping**
Add similar test as #10515 to the `HASH_CODE` operator in `REAL` type.

### CODE REVIEW

File: presto-main/src/test/java/com/facebook/presto/type/TestRealOperators.java [hunk 1]
```diff
@@ -245,4 +248,13 @@ public void testIndeterminate()
         assertOperator(INDETERMINATE, "cast(1.2 as real)", BOOLEAN, false);
         assertOperator(INDETERMINATE, "cast(123 as real)", BOOLEAN, false);
     }
+
+    @Test
+    public void testNanHash()
+    {
+        int nanValues[] = new int[] {floatToIntBits(Float.NaN), floatToRawIntBits(Float.NaN), -0x400000, 0x7fc00000};
```
Comment (findepi) [inline presto-main/src/test/java/com/facebook/presto/type/TestRealOperators.java]: - replace `-0x400000` with normal hex repr, `0xffc00000`
- `0x7fc00000 == floatToIntBits(Float.NaN) == floatToRawIntBits(Float.NaN)`, so its thrice on the list (i still see a value in having explicit hex and what `Float.NaN` yields). Perhaps you want something like 0x7fc01234
- `new int[]` is redundant

All in all, you could have something like this
```
int nanRepresentations[] = {floatToIntBits(Float.NaN), 0xffc00000, 0x7fc00000, 0x7fc01234, 0xffc01234};
for (int nanRepresentation : nanRepresentations) {
    assertTrue(isNaN(Float.intBitsToFloat(nanRepresentation)));
    assertEquals(RealOperators.hashCode(nanRepresentations[0]), RealOperators.hashCode(nanRepresentation));
}
```

File: presto-main/src/test/java/com/facebook/presto/type/TestRealOperators.java [hunk 2]
```diff
@@ -245,4 +248,13 @@ public void testIndeterminate()
         assertOperator(INDETERMINATE, "cast(1.2 as real)", BOOLEAN, false);
         assertOperator(INDETERMINATE, "cast(123 as real)", BOOLEAN, false);
     }
+
+    @Test
+    public void testNanHash()
```
Comment (findepi) [inline presto-main/src/test/java/com/facebook/presto/type/TestRealOperators.java]: can you name this method `testHash` and add some non-NaN cases as well?

can you also add similar test in `TestDoubleOperators` ?

Comment (jessesleeping) [reply presto-main/src/test/java/com/facebook/presto/type/TestRealOperators.java]: I am not sure how are we going to test non-Nan cases. It's hard to define correct behavior of `hash()`. We can check if there is no collision but that's not very informative. 
 
Do we want the output to be exactly the same as `REAL.hash()` which is a block+offset version defined in `RealType`? Or any other ideas?

Also, for `DoubleOperator`, it's native double type so we don't need to worry about `NaN`.

**Comment by jessesleeping** [comment]
I added test for `DOUBLE` as well.

**Pull request merged**